src/components/graph/graph.helper.js

```diff
@@ -43,18 +44,46 @@ const NODE_PROPS_WHITELIST = ['id', 'highlighted', 'x', 'y', 'index', 'vy', 'vx'
  * @param  {number} width - the width of the container area of the graph.
  * @param  {number} height - the height of the container area of the graph.
  * @param  {number} gravity - the force strength applied to the graph.
+ * @param  {Function} onCollide - this method takes a node as its param and returns a collision force radius.
  * @returns {Object} returns the simulation instance to be consumed.
  * @memberof Graph/helper
  */
-function _createForceSimulation(width, height, gravity) {
+function _createForceSimulation(width, height, gravity, onCollide) {
     const frx = d3ForceX(width / 2).strength(CONST.FORCE_X);
     const fry = d3ForceY(height / 2).strength(CONST.FORCE_Y);
     const forceStrength = gravity;
 
-    return d3ForceSimulation()
-        .force('charge', d3ForceManyBody().strength(forceStrength))
-        .force('x', frx)
-        .force('y', fry);
+    if (_isValidCollisionFunction(onCollide)) {
+        return (
+            d3ForceSimulation()
+                .force('charge', d3ForceManyBody().strength(forceStrength))
+                .force('x', frx)
+                .force('y', fry)
+                .force('y', fry)
+                // Each node from the data set is passed into the onCollide method
+                .force(
+                    'collide',
+                    d3ForceCollide(node => {
```

Comment:
It should be the client responsability to pass in the callback to `collide` force event, this way (wrapping the call of `onCollide` inside a call to `d3ForceCollide` we are killing the flexibility that we can provide.

IMO the client should do:

```javascript
// example, this fn should be passed into the config
const onCollide = d3.forceCollide().radius(function(d) { return d.r + 0.5; });
...
```

And then here `react-d3-graph` just does:

```javascript
...
.force('collide', onCollide);
...
```